posts/2025-12-16-project-goals-nov-update.md

```diff
@@ -0,0 +1,240 @@
+are available in its associated [tracking issue on the rust-project-goals
+repository](https://github.com/rust-lang/rust-project-goals/milestone/2).
+
+## Flagship goals
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/105'><strong>Bring the Async Rust experience closer to parity with sync Rust</strong></a></div>
+    <div style="flex: initial;"><progress value="21" max="40"></progress>
+</div>
+</div>
+
+Async closure stabilization has [been approved](https://github.com/rust-lang/rust/pull/132706), though the stabilization has not yet landed! The lang team ultimately opted to stabilize the trait name `AsyncFn` rather than the keyword-based `async Fn` syntax that was originally proposed. This decision came after discussion on the Flavors RFC which made it clear we were not at a consensus about whether the `async Trait` keyword would be used more generally or not. Given that, the team felt that the `AsyncFn` synta was a fine "next step". If we do ultimately adopt some form of `async Trait` keyword syntax, then `AsyncFn` can become a trait alias.
+
+Regarding return-type notation, an extension of return-type notation to cover Self::foo(..): Send [landed](https://github.com/rust-lang/rust/pull/129629) and we landed [#132047](https://github.com/rust-lang/rust/issues/132047) which fixes a known ICE. Stabilization PR is now unblocked.
+
+No major progress towards async drop reviews or team reorganization.
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/116'><strong>Resolve the biggest blockers to Linux building on stable Rust</strong></a></div>
+    <div style="flex: initial;"><progress value="29" max="55"></progress>
+</div>
+</div>
+
+This month saw steady progress on our checklist. dingxiangfei2009's PR renaming `derive(SmartPointer)` to `derive(CoercePointee)` was merged and he began the work to port the RFL codebase to use the new name. Alice Ryhl opened [RFC #3716](https://github.com/rust-lang/rfcs/pull/3716) proposing a way to manage compiler flags that alter the ABI and discussion (and some implementation work) has ensued. Finally, we landed [PR #119364](https://github.com/rust-lang/rust/issues/119364) making target blocks in asm-goto safe by default; this was based directly on experience from RFL which showed that [safe would be more useful]. We are still working to finalize another extension to asm-goto that arose from RFL requirements, [allowing `const` to support embedded pointers](https://github.com/rust-lang/rust/issues/128464). Finally we prepared [reference PR #1610](https://github.com/rust-lang/reference/issues/1610) describing the change to permit [Pointers to Statics in Constants](https://github.com/rust-lang/rust/issues/119618) that was stabilized last month.
```

Comment:
Sorry, should have reviewed this more closely. @nikomatsakis where is `[safe would be more useful]` supposed to link to?